lib/modules/coverage/index.js

```diff
@@ -42,6 +40,12 @@ class CodeCoverage {
     this.contractSources.parseSolcOutput(output);
   }
 
+  updateCoverageReport(cb) {
+    fs.mkdirpSync(this.dotEmbarkPath);
+    fs.writeFileSync(this.coverageReportPath, JSON.stringify(this.coverageReport));
```

Comment:
Since you have a callback, you can use the async version, that way the main process can continue while writing.